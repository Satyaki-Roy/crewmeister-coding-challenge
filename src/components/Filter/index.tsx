import * as React from 'react';
import {Box, InputLabel, MenuItem, FormControl} from '@mui/material';
import Select, {SelectChangeEvent} from '@mui/material/Select';

export interface Props {
  label: string;
  menuItems: string[];
  value: string;
  setValue: (value: string) => void
}

export default function Filter(props: React.PropsWithChildren<Props>): JSX.Element {
  (Filter as React.FC).displayName = 'Filter';
  const {label, menuItems, value, setValue} = props;

  // const [value, setValue] = React.useState<string>('');

  const handleChange = (event: SelectChangeEvent) => {
    setValue(event.target.value as string);
  };

  return (
    <Box sx={{width: 150, paddingTop: 1}}>
      <FormControl fullWidth>
        <InputLabel id="demo-simple-select-label">{label}</InputLabel>
        <Select
          labelId="demo-simple-select-label"
          id="demo-simple-select"
          value={value}
          label={label}
          onChange={handleChange}
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          {menuItems.map((e, index) => {
            return (
              <MenuItem key={index} value={e}>{e}</MenuItem>
            );
          })}
        </Select>
      </FormControl>
    </Box>
  );
}
