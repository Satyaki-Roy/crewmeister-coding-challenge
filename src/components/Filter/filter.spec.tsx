import React from 'react';
import {render} from '@testing-library/react';
import Filter from '.';

describe('Filter Component', () => {
  test('should match the snapshot', () => {
    const {container} = render(<Filter label="Absence Type" menuItems={['sickness', 'vacation']} value={""}
                                       setValue={() => {
                                       }}/>);
    expect(container).toMatchSnapshot();
  });

  test('should check the options', () => {
    const view = render(<Filter label="Absence Type" menuItems={['sickness', 'vacation']}
                                value={"sickness"}
                                setValue={() => {
                                }}/>);
    expect(view.container).toMatchSnapshot();
  })
})
