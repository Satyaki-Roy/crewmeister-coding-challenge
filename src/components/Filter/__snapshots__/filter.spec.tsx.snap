// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component should check the options 1`] = `
<div>
  <div
    class="MuiBox-root css-gzce58"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="demo-simple-select-label"
      >
        Absence Type
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          sickness
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="sickness"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-14lo706"
          >
            <span>
              Absence Type
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filter Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-gzce58"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="demo-simple-select-label"
      >
        Absence Type
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-yjsfm1"
          >
            <span>
              Absence Type
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
